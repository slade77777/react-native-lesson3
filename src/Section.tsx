import React from "react";
import {StyleSheet, Text, useColorScheme, View} from "react-native";
import {Colors} from "react-native/Libraries/NewAppScreen";

const Section: React.FC<{
    title?: string;
    description?: string;
}> = ({children, title, description}) => {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <View style={styles.sectionContainer}>
            <Text
                style={[
                    styles.sectionTitle,
                    {
                        fontSize: 30,
                        color: 'blue',
                    },
                ]}>
                {title}
            </Text>
            <Text
                style={[
                    styles.sectionTitle,
                    {
                        color: 'red'
                    },
                ]}>
                {description}
            </Text>
            <Text
                style={[
                    styles.sectionDescription,
                    {
                        color: isDarkMode ? Colors.light : Colors.dark,
                    },
                ]}>
                {children}
            </Text>
        </View>
    );
};


const styles = StyleSheet.create({
    sectionContainer: {
        marginTop: 32,
        paddingHorizontal: 24,
    },
    sectionTitle: {
        fontSize: 24,
        fontWeight: '600',
        color: 'pink'
    },
    sectionDescription: {
        marginTop: 8,
        fontSize: 18,
        fontWeight: '400',
    },
    highlight: {
        fontWeight: '700',
    },
});

export default React.memo(Section);
