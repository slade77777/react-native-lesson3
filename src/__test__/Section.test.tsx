import { render, fireEvent } from '@testing-library/react-native';
import Section from "../Section";
import React from "react";

test('test section', () => {
    const section = render(
        <Section description='description here' title='ahihi'/>
    )

    expect(section).toMatchSnapshot();
})

test('renderText correct', () => {
    const section = render(
        <Section description='description here' title='ahihi'/>
    )
    expect(section.getByText('ahihi')).toBeTruthy();
})
