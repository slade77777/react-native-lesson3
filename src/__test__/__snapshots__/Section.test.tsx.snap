// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test section 1`] = `
<View
  style={
    Object {
      "marginTop": 32,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "pink",
          "fontSize": 24,
          "fontWeight": "600",
        },
        Object {
          "color": "blue",
          "fontSize": 30,
        },
      ]
    }
  >
    ahihi
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "pink",
          "fontSize": 24,
          "fontWeight": "600",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    description here
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "400",
          "marginTop": 8,
        },
        Object {
          "color": "#444",
        },
      ]
    }
  />
</View>
`;
